### PR TITLE
Regional admin should have view and edit access to all regions 3224

### PR DESCRIPTION
**Pull request by ameliav**
#  #

## 🗣 Description ##
Regional admin should have view and edit access to all regions
CRASM-3224

## 💭 Motivation and context ##
Allow a regional admin to access multiple regions without interference from a global admin.

## 🧪 Testing ##
Acceptance Criteria

Login as regional admin have updated access to the following pages:
- VS Dashboard - all regions
- FIndings Library - all regions
- Manage Organization - all regions
- Manage Users - all regions
- User Registration - all regions
- No changes to other pages

## 📷 Screenshots (if appropriate) ##

Seeing all regions now as Regional Admin
image
image
image
image

Can see all regions but cannot see Admin Tools
image

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced in code comments.
- [ ] All relevant type-of-change labels have been added.
- [ ] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [ ] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.
- [ ] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).
- [ ] Create a pre-release (necessary if and only if the pre-release version was bumped).

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release (necessary if and only if the version was bumped).

### CODE REVIEW

File: backend/src/xfd_django/xfd_api/api_methods/organization.py
Comment (jayjaybunce) [inline backend/src/xfd_django/xfd_api/api_methods/organization.py:1]: Testing locally, in Manage Organizations a regional admin sees an empty table of the organizations where a global admin sees this table populated with the organizations. I tested all the other affected functionality and everything works outside of the /v2/organizations/search endpoint. See image for proposed fix

`/backend/src/xfd_django/xfd_api/api_methods/organizations.py`

Image

File: backend/src/xfd_django/xfd_api/tests/test_organization.py [hunk 1]
```diff
@@ -1831,8 +1831,8 @@ def test_add_user_to_org_v2_organization_not_found():
 
 
 @pytest.mark.django_db(transaction=True, databases=["default", "mini_data_lake"])
-def test_add_user_to_org_v2_region_mismatch():
```
Comment (cduhn17) [inline backend/src/xfd_django/xfd_api/tests/test_organization.py:1834]: Consider adding negative test to prove our authorization boundary/path, prevent privilege creep, and lock in the correct 401(authentication)/403(authorization) behavior.

File: backend/src/xfd_django/xfd_api/tests/test_user.py [hunk 2]
```diff
@@ -1479,7 +1480,7 @@ def test_update_user_v2_standard_user_cannot_update_own_email():
     )
```
Comment (cduhn17) [inline backend/src/xfd_django/xfd_api/tests/test_user.py:1480]: Consider adding negative test to prove our authorization boundary/path, prevent privilege creep, and lock in the correct 401(authentication)/403(authorization) behavior.

File: frontend/src/pages/Users/Users.tsx [hunk 3]
```diff
@@ -104,10 +104,7 @@ export const Users: React.FC = () => {
         row.full_name = `${row.first_name} ${row.last_name}`;
       });
```
Comment (cduhn17) [inline frontend/src/pages/Users/Users.tsx:107]: Add a Playwright spec that:
- Logs in as Regional Admin
- Tries to hit the Admin Tools path by URL
- Expects redirect or 403 page

Files:
frontend/src/pages/Users/Users.tsx and related guards
A Playwright spec under playwright/e2e/role-regional-admin.spec.ts confirming the bounce

**Review by cduhn17 — COMMENTED** [review]
See comments.

**Review by rapidray12 — COMMENTED** [review]
Do we have the corresponding checks in the backend?

File: frontend/src/pages/Users/Users.tsx (hunk 3 above)
Comment (ameliav) [reply frontend/src/pages/Users/Users.tsx:107]: A ticket was created to make playwright tests specifically for Regional Admin access. I've added a comment to the ticket for a request to make this specific test here: https://maestro.dhs.gov/jira/browse/CRASM-3183

File: backend/src/xfd_django/xfd_api/tests/test_user.py (hunk 2 above)
Comment (ameliav) [reply backend/src/xfd_django/xfd_api/tests/test_user.py]: Thanks for the consideration. Looks like there's already negative tests covering authorization and authentication boundaries.
Examples:
- test_update_user_v2_as_standard_user_fails (403 unauthorized access)
- test_update_user_v2_no_auth (401 unauthenticated)
- test_update_user_v2_non_existent_user (404 not found)
- test_update_user_v2_update_userType_by_non_admin_fails (403 only global admins can update user type)
- and more

Let me know if you had anything specific in mind to add.

File: backend/src/xfd_django/xfd_api/tests/test_organization.py (hunk 1 above)
Comment (ameliav) [reply backend/src/xfd_django/xfd_api/tests/test_organization.py]: Similarily to test_user, I see a few negative tests already in place, but let me know if you had something specific in mind. Thanks.

File: frontend/src/pages/Users/Users.tsx (hunk 3 above)
Comment (cduhn17) [reply frontend/src/pages/Users/Users.tsx:107]: Does this create a dependency with the advent of the new ticket therefore putting this PR on hold? Looping in Ray and Stephen @rapidray12 and @schmelz21

Comment (ameliav) [reply frontend/src/pages/Users/Users.tsx]: As discussed, the new ticket depends on this one as this updates the regional admin permissions. A TODO comment has been added here: [e3323b8](https://github.com/cisagov/XFD/pull/1283/commits/e3323b8ee080595b98dbe3ed1df0e18855d3a9dd)

Comment (schmelz21) [reply frontend/src/pages/Users/Users.tsx:107]: @cduhn17 - thanks for the check. The TODO will be added right after this merge. It is currently being worked on and will be subsequently added right after. This will allow us to smoke test in develop while the playwright baseline code is created for this usertype. There is a fundamental set of code/ticket that was just approved to be worked on while this sat in review.

**Review by jsalinasnttdata — DISMISSED** [review]
Frontend logic looks good. I also was able to test locally without any errors.

File: backend/src/xfd_django/xfd_api/tests/test_user.py (hunk 2 above)
Comment (cduhn17) [reply backend/src/xfd_django/xfd_api/tests/test_user.py:1480]: see note from stephen 10/21

File: backend/src/xfd_django/xfd_api/tests/test_organization.py (hunk 1 above)
Comment (cduhn17) [reply backend/src/xfd_django/xfd_api/tests/test_organization.py:1834]: see note from stephen 10/21

File: frontend/src/pages/Users/Users.tsx (hunk 3 above)
Comment (cduhn17) [reply frontend/src/pages/Users/Users.tsx:107]: Ok. I referenced your last comment and closed the conversations

**Review by cduhn17 — DISMISSED** [review]
LGTM - see note from stephen 10/21 in regards to TODO requests and future tickets.

**Review by rapidray12 — DISMISSED** [review]
Lots of changes.  It will require significant testing

**Comment by ameliav** [comment]
Spoke to Jack about removing the TODO comment once he starts work on the regional admin playwright tests CRASM-3183. He will need to pull this branch into his or wait for this branch to merge into develop to start work on it.

**Review by hawkishpolicy — APPROVED** [review]
Works!

LGTM.

**Review by jsalinasnttdata — APPROVED** [review]
